build-js.sh

```diff
@@ -3,12 +3,113 @@
 # to run this, as the builds are bundled in the repo in bin/. Running this is
 # useful if you are a developer and want to update those builds.
 #
+# Usage: build-js.sh
+# Usage: EMSCRIPTEN=path/to/emscripten build-js.sh  # explicit emscripten dir
+#
+# Emscripten's em++ and tools/webidl_binder.py will be accessed through the
+# env var EMSCRIPTEN, e.g. ${EMSCRIPTEN}/em++
+#
+# You can get emscripten from
+# http://kripken.github.io/emscripten-site/docs/getting_started/downloads.html
+#
+set -e
+
+if [ "$1" == "-h" ] || [ "$1" == "--help" ] || [ "$1" == "-help" ]; then
+  echo "usage: $0 [-g]" >&2
+  echo "-g  produce debug build" >&2
+  exit 1
+fi
+
+if [ "$EMSCRIPTEN" == "" ]; then
+  echo "$0: EMSCRIPTEN environment variable is not set" >&2
+  exit 1
+fi
+
+if [ ! -d "$EMSCRIPTEN" ]; then
+  echo "$0: $EMSCRIPTEN is not a directory" >&2
+  exit 1
+fi
+
+EMCC_ARGS="-std=c++11 --memory-init-file 0"
+EMCC_ARGS="$EMCC_ARGS -s ALLOW_MEMORY_GROWTH=1"
+EMCC_ARGS="$EMCC_ARGS -s DEMANGLE_SUPPORT=1"
+OUT_FILE_SUFFIX=
+
+if [ "$1" == "-g" ]; then
+  EMCC_ARGS="$EMCC_ARGS -O0"
+  EMCC_ARGS="$EMCC_ARGS --llvm-opts 0 --llvm-lto 0"
+  EMCC_ARGS="$EMCC_ARGS -profiling"
+  OUT_FILE_SUFFIX=-g
+else
+  EMCC_ARGS="$EMCC_ARGS -Oz"
```

Comment:
@kripken Disabling optimizations for the debug build makes it easier to look through the generated code and to draw parallels to the source IMHO. Happy to change it to `-Oz` for both targets if you prefer. Let me know!